### PR TITLE
feat: added support to remote files

### PR DESCRIPTION
**Pull request by TheGhoul21**
- Sound.js now checks if is RelativePath using also http and https
- RNSoundModule now uses setAudioStreamType(AudioManager.STREAM_MUSIC)
to stream music
- Exception now are handled differently

### CODE REVIEW

**Comment by TheGhoul21** [comment]
#134

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 1]
```diff
@@ -56,6 +66,19 @@ protected MediaPlayer createMediaPlayer(final String fileName) {
...
     }
+    if(fileName.startsWith("http://") || fileName.startsWith("https://")) {
+      MediaPlayer mediaPlayer = new MediaPlayer();
+      mediaPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC);
+      Log.i("RNSoundModule", fileName);
+      try {
+        mediaPlayer.setDataSource(fileName);
+      } catch(IOException e) {
+        Log.e("RNSoundModule", "Exception", e);
+        return null;
```
Comment (benvium) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: Nice - streaming is a good idea over http for sure.

Comment (benvium) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: Did you have a specific reason not to invoke a callback error when the setDataSource throws an exception?

Comment (TheGhoul21) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: it usually throws an exception when internet is missing, or when the file does not exist. Do you suggest to handle differently those two? (returning null here, will invoke the error callback anyway later on with a more generic "Resouce not found" error)

Comment (benvium) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: @TheGhoul21 ah - thanks, I didn't notice that it'd hit the error callback anyway. Thanks for the clarification. This seems a sensible solution

**Comment by matthieugayon** [comment]
Any chance this gets merged soon ?

**Comment by benvium** [comment]
@matthieugayon Let me know if it works for you, and if so we should merge it.

**Comment by matthieugayon** [comment]
Ok I'll try to get back to you asap on that, most probably tomorrow

**Comment by ndbroadbent** [comment]
@TheGhoul21 can I ask, do you know what kind of caching would be happening here? Would Android be caching the data at all, or redownloading for each new `mediaPlayer`?

The reason I'm asking: I want to build "code push" feature for my app's sounds, so that I can add or update sounds without needing to push a new version to the app store. I'm working on a game, so it would be amazing if I could immediately push new levels + images + sounds.

So I was just wondering if it would be easy to add some cache controls. If not, I can just handle the requests myself and play the sounds from a local directory.

**Comment by GShotwell** [comment]
+1 for this getting merged, it's the missing link for react native audio playing

**Comment by ndbroadbent** [comment]
This is actually great timing. As I mentioned, I've been setting up CodePush, and switching to React Native's asset manager. So I just realized that I actually do need this, because now I get a URL during development:

> "http://10.0.3.2:8081/assets/src/lib/sounds/alert.mp3?platform=android&hash=442aeb48612c0aed468b12f4e2ca01bc"

It also might be good to add "[React Native Static Resources](https://facebook.github.io/react-native/docs/images.html#static-non-image-resources)" to the feature matrix in the README, maybe with a link to that page.

**Comment by SteffeyDev** [comment]
I just replaced react-native-sound with TheGhoul21/react-native-sound in my cross platform app and can confirm that that Android load from network works identical to iOS.  It should probably be in the readme somewhere that the syntax to load from network is 
`var audio = new Sound(networkURL, null, (error) => {});`

**Comment by benvium** [comment]
@SteffeyDev Thanks for the confirmation. I've merged it now. Great job @TheGhoul21

**Pull request merged**